src/java/voldemort/server/protocol/admin/AdminServiceRequestHandler.java

```diff
@@ -1943,10 +1943,25 @@ private Message handleDisableStoreVersion(VAdminProto.DisableStoreVersionRequest
         }
 
         if (response.getDisableSuccess()) {
-            // Then we also want to put the server in offline mode
+            // Voldemort Client can detect and handle failure only at a server
+            // level. Disabling a store inside Server is not possible . Client
+            // creates connection to the server and shares the same connection
+            // for multiple stores. Client failure detection is based on this
+            // connection error. So entire server will be marked as offline for
+            // now.
+
+            boolean roFetchState = metadataStore.getReadOnlyFetchEnabledUnlocked();
+
             VAdminProto.SetOfflineStateRequest offlineStateRequest =
                     VAdminProto.SetOfflineStateRequest.newBuilder().setOfflineMode(true).build();
             handleSetOfflineState(offlineStateRequest);
```

Comment:
There is no convenient method for just putting the state in offline. 

handleSetOffline needs to pass it all the way down (VoldemortServer to MetadataStore) and add new parameters for setting the individual states.This will make the fix little more complicated. I would like to leave it in this state for now.